WeakAuras/RegionTypes/Icon.lua

```diff
@@ -525,9 +527,10 @@ local function modify(parent, region, data)
     end
 
     function region:SetTime(duration, expirationTime)
-      if (duration > 0) then
+      local adjustedDuration = region.progressScale or duration;
+      if (adjustedDuration > 0) then
         cooldown:Show();
-        cooldown:SetCooldown(expirationTime - duration, duration);
+        cooldown:SetCooldown(expirationTime - adjustedDuration, adjustedDuration);
```

Comment:
I did. It does. Both normal and inverse. But I can remove it if it's risky to rely on that.